### PR TITLE
fix(Core): Impelement CREATURE_FLAG_EXTRA_NO_PLAYER_DAMAGE_REQ

### PR DESCRIPTION
**Pull request by Krejza9**
NPCs can help with killing this creature and player will still be credited if he tags the creature
## Changes Proposed:
-  Implement CREATURE_FLAG_EXTRA_NO_PLAYER_DAMAGE_REQ

## SOURCE:

TC

## Tests Performed:

- Build

## How to Test the Changes:

1. Find creature with flags_extra 2097152
2.  Aggro creature with help of NPC
3. Don't give him any dmg
4. After a friendly NPC kills him, you get XP

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Miscellaneous/Formulas.cpp [hunk 1]
```diff
@@ -84,6 +84,13 @@ uint32 Acore::XP::Gain(Player* player, Unit* unit, bool isBattleGround /*= false
         }
 
         xpMod *= isBattleGround ? sWorld->getRate(RATE_XP_BG_KILL) : sWorld->getRate(RATE_XP_KILL);
+       
+        // if players dealt less than 50% of the damage and were credited anyway (due to CREATURE_FLAG_EXTRA_NO_PLAYER_DAMAGE_REQ), scale XP gained appropriately (linear scaling)
+        if (creature && creature->m_PlayerDamageReq)
+        {
+            xpMod *= 1.0f - 2.0f * creature->m_PlayerDamageReq / creature->GetMaxHealth();
+        }
+
```
Comment (FrancescoBorzi) [inline src/server/game/Miscellaneous/Formulas.cpp:93]: @Krejza9 you got whitespaces on lines 87 and 93, that's why the codestyle check is failing in the CI

Comment (Krejza9) [reply src/server/game/Miscellaneous/Formulas.cpp]: done

**Comment by pklloveyou** [comment]
I put this PR on the server

**Review by Helias — APPROVED** [review]

**Comment by pklloveyou** [comment]
So far, there are no player feedback errors

**Comment by pklloveyou** [comment]
So far, there are no player feedback errors

**Comment by pklloveyou** [comment]
So far, there are no player feedback errors.

**Pull request merged**